rest_framework_json_api/renderers.py

```diff
@@ -111,7 +111,8 @@ def extract_relationships(cls, fields, resource, resource_instance):
                     relation_data.append(
                         OrderedDict([
                             ('type', relation_type),
-                            ('id', encoding.force_text(related_object.pk))
+                            ('id', encoding.force_text(getattr(
```

Comment:
What's happening with this `getattr`? If my understanding of `getattr` is correct, then something seems off here. Let's say `fields.lookup_field = 'foo'`. Wouldn't this `getattr` call be equivalent to `related_object.pk.foo`?

Reply:
something went wrong when cherry-picking, sorted now